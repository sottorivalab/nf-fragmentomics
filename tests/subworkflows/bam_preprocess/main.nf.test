nextflow_workflow {

    name "Test Workflow BAM_PREPROCESS"
    script "subworkflows/bam_preprocess/main.nf"
    workflow "BAM_PREPROCESS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                    null
                ]
                input[1] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr22/sequence/chr22_23800000-23980000.2bit', checkIfExists: true)
                ]
                input[2] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.blacklist_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}
