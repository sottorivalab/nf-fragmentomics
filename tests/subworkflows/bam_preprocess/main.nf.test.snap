{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.gc_correct.bw:md5,936b325a0eec195768a575c3f439e4d6"
                    ]
                ],
                "wiggle_ch": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.gc_correct.bw:md5,936b325a0eec195768a575c3f439e4d6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T14:47:43.663956"
    }
}